src/pages/transactions/components/TransactionMatchDetails.tsx

```diff
@@ -0,0 +1,191 @@
+/**
+ * Invoice Ninja (https://invoiceninja.com).
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2022. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+import {
+  Dispatch,
+  FormEvent,
+  ReactNode,
+  SetStateAction,
+  useEffect,
+  useState,
+} from 'react';
+import { toast } from 'common/helpers/toast/toast';
+import { request } from 'common/helpers/request';
+import { endpoint } from 'common/helpers';
+import { AxiosError } from 'axios';
+import { useQueryClient } from 'react-query';
+import { TransactionStatus } from 'common/enums/transactions';
+import { ConvertButton } from './ConvertButton';
+import { route } from 'common/helpers/route';
+import { TransactionResponse } from 'common/interfaces/transactions';
+import { GenericSingleResourceResponse } from 'common/interfaces/generic-api-response';
+import { ListBox } from './ListBox';
+
+export interface TransactionDetails {
+  base_type: string;
+  transaction_id: string;
+  status_id: string;
+}
+
+interface Props {
+  transactionDetails: TransactionDetails;
+  isCreditTransactionType: boolean;
+  setActionButton: Dispatch<SetStateAction<ReactNode>>;
+}
+
+export function TransactionMatchDetails(props: Props) {
+  const queryClient = useQueryClient();
+
+  const [isFormBusy, setIsFormBusy] = useState<boolean>(false);
+
+  const [invoiceIds, setInvoiceIds] = useState<string[]>();
+
+  const [vendorIds, setVendorIds] = useState<string[]>();
+
+  const [expenseCategoryIds, setExpenseCategoryIds] = useState<string[]>();
+
+  const convertToPayment = (event: FormEvent<HTMLFormElement>) => {
```

Comment:
We rarely wanna expose direct events. Any chance you can make this function accepts the data instead of global event?

Reply:
Actually the data is within the component so we don't have to pass it `const [invoiceIds, setInvoiceIds] = useState<string[]>();` and then I use it later within in the function: `invoiceIds.join(','),`. Event here is just to prevent default.